pkg/daemon/daemon.go

```diff
@@ -117,27 +117,36 @@ func New(
 	return dn, nil
 }
 
-// Run first does state checking for the node, and then blocks. Updates are
-// notified via an informer. Successful updates shouldn't return, and should
-// just reboot the node. Errors during state checks and updates are sent up
-// the call stack.
-//
-// The flow goes something like this -
-// 1. Sanity check if we're in a degraded state. If yes, handle appropriately.
-// 2. we restarted for some reason. the happy path reason we restarted is
-//    because of a machine reboot. validate the current machine state is the
-//    desired machine state. if we aren't try updating again. if we are, update
-//    the current state annotation accordingly.
-// 3. set up the informer, and block waiting for updates. Updates are handled
-//    by callback functions.
-// 4. if an update is requested by the controller, we assume that that means
-//    something changed and apply the desired config no matter what.
+// Run finishes informer setup and then blocks, and the informer will be
+// responsible for triggering callbacks to handle updates. Successful
+// updates shouldn't return, and should just reboot the node.
 func (dn *Daemon) Run(stop <-chan struct{}) error {
 	defer utilruntime.HandleCrash()
 
 	glog.Info("Starting MachineConfigDaemon")
 	defer glog.Info("Shutting down MachineConfigDaemon")
```

Comment:
Minor: at this point, we might as well move these to `runStartCmd` before we call `CheckStateOnBoot`?

Reply:
Good point, will fix